File: surf-data-api/src/main/java/dev/slne/surf/data/api/config/auto/AdventureAutoConfiguration.java
package dev.slne.surf.data.api.config.auto;

import net.kyori.adventure.text.minimessage.MiniMessage;
import org.springframework.boot.autoconfigure.AutoConfiguration;
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
import org.springframework.context.annotation.Bean;

@AutoConfiguration
@ConditionalOnClass(MiniMessage.class)
public class AdventureAutoConfiguration {

  @Bean
  public MiniMessage miniMessage() {
    return MiniMessage.miniMessage();
  }
}

File: surf-cloud-api/surf-cloud-api-common/src/main/kotlin/dev/slne/surf/cloud/api/common/config/auto/AdventureAutoConfiguration.kt
package dev.slne.surf.cloud.api.common.config.auto

import net.kyori.adventure.text.minimessage.MiniMessage
import org.springframework.boot.autoconfigure.AutoConfiguration
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass
import org.springframework.context.annotation.Bean

@AutoConfiguration
@ConditionalOnClass(MiniMessage::class)
class AdventureAutoConfiguration {
    @Bean
    fun miniMessage(): MiniMessage {
        return MiniMessage.miniMessage()
    }
}
